edu_edfi_airflow/dags/edfi_resource_dag.py

```diff
@@ -48,116 +63,155 @@ def __init__(self,
 
         pool     : str,
         tmp_dir  : str,
-
         multiyear: bool = False,
         schedule_interval_full_refresh: Optional[str] = None,
 
         use_change_version: bool = True,
+        get_key_changes: bool = False,
+        run_type: str = "default",
+        resource_configs: Optional[List[dict]] = None,
+        descriptor_configs: Optional[List[dict]] = None,
+
         change_version_table: str = '_meta_change_versions',
+        deletes_table: str = '_deletes',
+        key_changes_table: str = '_key_changes',
+        descriptors_table: str = '_descriptors',
 
-        slack_conn_id: str = None,
-        dbt_incrementer_var: str = None,
+        dbt_incrementer_var: Optional[str] = None,
 
         **kwargs
     ) -> None:
+        self.run_type = run_type
+        self.use_change_version = use_change_version
+        self.get_key_changes = get_key_changes
+
         self.tenant_code = tenant_code
         self.api_year = api_year
-        self.multiyear = multiyear
-        self.schedule_interval_full_refresh = schedule_interval_full_refresh  # Force full-refresh on a scheduled cadence
 
         self.edfi_conn_id = edfi_conn_id
         self.s3_conn_id = s3_conn_id
         self.snowflake_conn_id = snowflake_conn_id
-        self.slack_conn_id = slack_conn_id
 
         self.pool = pool
         self.tmp_dir = tmp_dir
+        self.multiyear = multiyear
+        self.schedule_interval_full_refresh = schedule_interval_full_refresh  # Force full-refresh on a scheduled cadence
 
-        self.use_change_version = use_change_version
         self.change_version_table = change_version_table
+        self.deletes_table = deletes_table
+        self.key_changes_table = key_changes_table
+        self.descriptors_table = descriptors_table
 
         self.dbt_incrementer_var = dbt_incrementer_var
+        
+        ### Parse optional config objects (improved performance over adding resources manually).
```

Comment:
In our current DAG-initialization code, resources and deletes are added individually to the DAG. This is both inefficient and unnecessary. This refactor allows the Ed-Fi resources and descriptors objects to be passed directly during init, and these are looped over internally.

Method `parse_endpoint_configs()` takes either our YAML structure or a list of `(namespace, endpoint)` tuples (the output of `EdFiClient.resources`) and prepares a list of config bundles.